scipy/optimize/_minimize_constrained.py

```diff
@@ -0,0 +1,485 @@
+from warnings import warn
+import time
+from .optimize import OptimizeResult
+
+
+TERMINATION_MESSAGES = {
+    0: "The maximum number of function evaluations is exceeded.",
+    1: "`gtol` termination condition is satisfied.",
+    2: "`xtol` termination condition is satisfied.",
+    3: "`callback` function requested termination"
+}
+
+
+class sqp_printer:
+    @staticmethod
+    def print_header():
+        print("|{0:^7}|{1:^7}|{2:^7}|{3:^10}|{4:^10}|{5:^10}|{6:^10}|"
+              .format("niter", "f evals", "CG iter", "tr radius", "penalty",
+                      "opt", "c viol"))
+        s = "-"*6 + ":"
+        s2 = ":" + "-"*8 + ":"
+        print("|{0:^7}|{1:^7}|{2:^7}|{3:^10}|{4:^10}|{5:^10}|{6:^10}|"
+              .format(s, s, s, s2, s2, s2, s2))
+
+    @staticmethod
+    def print_problem_iter(niter, nfev, cg_niter, tr_radius,
+                           penalty, opt, c_viol):
+        print("|{0:>7}|{1:>7}|{2:>7}| {3:^1.2e} | {4:^1.2e} |"
+              " {5:^1.2e} | {6:^1.2e} |"
+              .format(niter, nfev, cg_niter, tr_radius, penalty,
+                      opt, c_viol))
+
+    @staticmethod
+    def print_footer():
+        print("")
+        print((7*3 + 10*4 + 8)*"-")
+        print("")
+
+
+class ip_printer:
+    @staticmethod
+    def print_header():
+        print("|{0:^7}|{1:^7}|{2:^7}|{3:^13}|{4:^10}|{5:^10}|{6:^10}|{7:^10}|"
+              .format("niter", "f evals", "CG iter", "barrier param",
+                      "tr radius", "penalty", "opt", "c viol"))
+        s = "-"*6 + ":"
+        s2 = ":" + "-"*11 + ":"
+        s3 = ":" + "-"*8 + ":"
+        print("|{0:^7}|{1:^7}|{2:^7}|{3:^13}|{4:^10}|{5:^10}|{6:^10}|{7:^10}|"
+              .format(s, s, s, s2, s3, s3, s3, s3))
+
+    @staticmethod
+    def print_problem_iter(niter, nfev, cg_niter, barrier_parameter, tr_radius,
+                           penalty, opt, c_viol):
+        print("|{0:>7}|{1:>7}|{2:>7}|   {3:^1.2e}  | "
+              "{4:^1.2e} | {5:^1.2e} | {6:^1.2e} | {7:^1.2e} |"
+              .format(niter, nfev, cg_niter, barrier_parameter,
+                      tr_radius, penalty, opt, c_viol))
+
+    @staticmethod
+    def print_footer():
+        print("")
+        print((7*3 + 13 + 10*4 + 9)*"-")
+        print("")
+
+
+def minimize_constrained(fun, x0, grad, hess=None, constraints=(),
+                         method=None, xtol=1e-8, gtol=1e-8,
+                         sparse_jacobian=None, options={},
+                         callback=None, max_iter=1000,
+                         verbose=0):
+    """Minimize scalar function subject to constraints.
+
+    Parameters
+    ----------
+    fun : callable
+        The objective function to be minimized.
+
+            fun(x) -> float
+
+        where x is an array with shape (n,).
+    grad : callable
+        Gradient of the objective function:
+
+            grad(x) -> array_like, shape (n,)
+
+        where x is an array with shape (n,).
+    hess : {callable, None}, optional
```

Comment:
When None is provided it consider the hessian is a matrix filled with zeros. I will include some explanation on the documentation.